tests/lib/prepare-restore.sh

```diff
@@ -293,7 +293,11 @@ prepare_project() {
     fi
 
     if [ "$SPREAD_BACKEND" = "testflinger" ]; then
-        adduser --uid 12345 --extrausers --quiet --disabled-password --gecos '' test
+        if os.query is-core-ge 24; then
+            useradd --uid 12345 -m --extrausers test
+        else
+            adduser --uid 12345 --extrausers --quiet --disabled-password --gecos '' test
+        fi
```

Comment:
Is the `--disabled-password` essential, and is this the default behavior for `useradd`? I don't see any equivalent option there.

Also, the `--gecos` option is deprecated, and `--comment` should be used instead. But if the comment is empty, is `--gecos` even necessary?

```
       --comment comment
              Set the comment field for the new entry generated.  adduser will not ask for the information  if  this  option  is
              given.  This field is also known under the name GECOS field and contains information that is used by the finger(1)
              command.   This  used  to  be  the  --gecos option, which is deprecated and will be removed after Debian bookworm.
```